CI/package-deps-windows.ps1

```diff
@@ -0,0 +1,159 @@
+
+function Package-OBS-Deps-Main {
+    $CheckoutDir = git rev-parse --show-toplevel
+    $ProductName = "windows-deps"
+    $DepsBuildDir = "${CheckoutDir}\windows\obs-win-deps"
+
+    . "${CheckoutDir}\CI\include\build_support_windows.ps1"
+
+    Write-Status "Create Windows OBS deps package"
+
+    Ensure-Directory "${CheckoutDir}"
+
+    if (!$CurrentDate) {
+        $CurrentDate = Get-Date -UFormat "%Y-%m-%d"
+    }
+    $FileName = "${ProductName}-${CurrentDate}.tar.gz"
+    $CrossDirBase = "${CheckoutDir}\windows\obs-cross-deps"
+    $NativeDirBase = ""
+
+    if (Test-Path "${CheckoutDir}\windows\obs-native-deps" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows\obs-native-deps"
+    } elseif (Test-Path "${CheckoutDir}\windows_native_build_temp" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows_native_build_temp"
+    }
+
+    if (!(Test-Path "${CrossDirBase}\x86" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86"
+    } elseif (!(Test-Path "${CrossDirBase}\x86_64" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86_64"
+    } elseif (!(Test-Path "${NativeDirBase}\win32" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win32"
+    } elseif (!(Test-Path "${NativeDirBase}\win64" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win64"
+    }
+
+    Remove-ItemIfExists "${DepsBuildDir}"
+    Ensure-Directory "${DepsBuildDir}\win32"
+    Ensure-Directory "${DepsBuildDir}\win64"
+    cd ..
+
+    $Packages = @(
+        [pscustomobject]@{Arch='x86'; WinArch='win32'}
+        [pscustomobject]@{Arch='x86_64'; WinArch='win64'}
+    )
+    Foreach ($Package in $Packages) {
+        if ($Package.Arch -eq "x86") {
+            $CrossDir = "${CrossDirBase}\x86"
+            $NativeDir = "${NativeDirBase}\win32"
+            $FinalDir = "${DepsBuildDir}\win32"
+        } elseif ($Package.Arch -eq "x86_64") {
+            $CrossDir = "${CrossDirBase}\x86_64"
+            $NativeDir = "${NativeDirBase}\win64"
+            $FinalDir = "${DepsBuildDir}\win64"
+        }
+
+        Write-Step "Copy $($Package.WinArch)/$($Package.Arch) files..."
+
+        # Copy cross-compiled deps first
+        Write-Step "Copy cross-compiled $($Package.WinArch)/$($Package.Arch) files..."
+        Copy-Item -Path "${CrossDir}\bin" -Destination "${FinalDir}" -Recurse
+        Copy-Item -Path "${CrossDir}\include" -Destination "${FinalDir}" -Recurse
+        Copy-Item -Path "${CrossDir}\licenses" -Destination "${FinalDir}" -Recurse
+
+        Write-Step "Clean up cross-compiled $($Package.WinArch)/$($Package.Arch) files..."
+        # Remove unneeded files before copying native-compiled deps
+        # Make sure symlinks still exist before trying to remove them
+        Remove-Item -Path "${FinalDir}\bin\libmbedtls.dll" -Force
```

Comment:
I wonder if it might be simpler to add the file paths to an array and just pipe it to `Remove-Item`:

```
@(
    "${FinalDir}\bin\libmbedtls.dll",
    "${FinalDir}\bin\libmbedx509.dll",
    [...]
) | Remove-ItemIfExists -Force
```

Reply:
More PowerShell-esque, perhaps.  Simpler?  Debatable.

For now, I'm content to leave it as-is since it works and leave this question to a future change.

Reply:
My idea was that adding additional entries to remove is simpler via an additional entry in a array instead of adding a whole function call.

Reply:
> My idea was that adding additional entries to remove is simpler via an additional entry in a array instead of adding a whole function call.

That particular aspect may be simpler, yes.  I'm fine with that in general, just perhaps for a future iteration unless I get to it very quickly.